src/coreclr/vm/codeman.cpp

```diff
@@ -1290,22 +1290,7 @@ void EEJitManager::SetCpuInfo()
 #else
     _ASSERTE((cpuFeatures & XArchIntrinsicConstants_VectorT128) != 0);
 #endif
-
```

Comment:
I think we want to keep this hear so that it can feed into EnsureValidInstructionSetSupport. `EnsureValidInstructionSetSupport` operates on the `VectorT...` bits.

Reply:
I meant to only delete `CPUCompileFlags.Clear(InstructionSet_VectorT128);` from here and add the following right after the `EnsureValidInstructionSetSupport` call (or even make it part of the auto-generated EnsureValidInstructionSetSupport call):

```
if (CPUCompileFlags.IsSet(InstructionSet_VectorT512))
{
    CPUCompileFlags.Clear(InstructionSet_VectorT256);
    CPUCompileFlags.Clear(InstructionSet_VectorT128);
    CPUCompileFlags.Clear(InstructionSet_VectorT64);
}
else if (CPUCompileFlags.IsSet(InstructionSet_VectorT256))
{
    CPUCompileFlags.Clear(InstructionSet_VectorT128);
    CPUCompileFlags.Clear(InstructionSet_VectorT64);
}
else if (CPUCompileFlags.IsSet(InstructionSet_VectorT128))
{
    CPUCompileFlags.Clear(InstructionSet_VectorT64);
}
```

Reply:
So was my initial commit correct?

Reply:
> So was my initial commit correct?

The initial commit had two issues:
- It was modifying auto-generated code, without matching change in the auto-generator.
- It was setting the bits in `EnsureValidInstructionSetSupport`. `EnsureValidInstructionSetSupport` should be only clearing the bits to get to a valid combination.

Reply:
Thanks, now I see what you meant - changed!

Reply:
note: `InstructionSet_VectorT64` doesn't exist (even on arm) so I didn't copy it

Reply:
Yep, no VectorT64 since the underlying Vector<T> has always been at least 16-bytes. We could relax that, it would just be more work.